### PR TITLE
feat: clean-up resharding v2: part 1

### PR DESCRIPTION
**Pull request by Trisfald**
First part of cleanup of legacy resharding V2 code.

**Notes**: 
- a subset of resharding V2 will remain as it is required by the `resharding` tool
    - resharding V2 source files have now the `_v2` suffix
- for better user experience no changes will be made to resharding config

This is just the beginning.. many other things must be removed as well:
- `StateSnapshotType::ForReshardingOnly`
- `StateChangeCauseView::Resharding`
- `StateChangesForResharding`
- `InvalidSplitShardsIds`
- `ShardSyncStatus::{ReshardingScheduling, ReshardingApplying}` 
- `ReshardingResult`
- All resharding metrics
- `ApplyResultForResharding`
- `State_sync::resharding_state_roots`
- Various resharding fn in `chain.rs`
- Resharding handle
- `ShardUpdateReason::Resharding`
- Reassigning of outgoing receipt in `chain/src/store/mod.rs`

I think it's better to tackle the other items in different PRs, but I don't have a strong opinion

### CODE REVIEW

File: chain/chunks/src/client.rs [hunk 1]
```diff
@@ -115,57 +114,12 @@ impl ShardedTransactionPool {
         }
         reintroduced_count
     }
-
-    /// Migrate all of the transactions in the pool from the old shard layout to
-    /// the new shard layout.
-    /// It works by emptying the pools for old shard uids and re-inserting the
-    /// transactions back to the pool with the new shard uids.
-    pub fn reshard(&mut self, old_shard_layout: &ShardLayout, new_shard_layout: &ShardLayout) {
```
Comment (wacban) [inline chain/chunks/src/client.rs:123]: It feels like we will need transaction pool resharding for IR too.

File: chain/client/src/client.rs [hunk 2]
```diff
@@ -1640,27 +1639,6 @@ impl Client {
             if let Err(err) = self.send_network_chain_info() {
                 error!(target: "client", ?err, "Failed to update network chain info");
             }
-
-            // If the next block is the first of the next epoch and the shard
-            // layout is changing we need to reshard the transaction pool.
-            // TODO make sure transactions don't get added for the old shard
-            // layout after the pool resharding
```
Comment (wacban) [inline chain/client/src/client.rs:1647]: ditto - it's likely still needed for IR. Maybe even without changes but needs checking. I'll add an item to look into that to the tracking issue.

File: chain/client/src/sync/state.rs [hunk 3]
```diff
@@ -319,24 +317,10 @@ impl StateSync {
...
-                    debug_assert!(need_to_reshard);
-                    self.sync_shards_resharding_scheduling_status(
-                        shard_id,
-                        shard_sync_download,
-                        sync_hash,
-                        chain,
-                        resharding_scheduler,
-                        me,
-                    )?;
+                    panic!("Resharding V2 scheduling is no longer supported")
```
Comment (wacban) [inline chain/client/src/sync/state.rs:320]: Are you planning to follow up and delete those entirely?

File: integration-tests/src/tests/client/resharding_v2.rs [hunk 4]
```diff
@@ -0,0 +1,925 @@
...
+
+#[test]
+fn test_latest_protocol_missing_chunks_mid_missing_prob() {
+    test_latest_protocol_missing_chunks(0.5, 26);
+}
+
+#[test]
+fn test_latest_protocol_missing_chunks_high_missing_prob() {
+    test_latest_protocol_missing_chunks(0.9, 27);
+}
```
Comment (wacban) [inline integration-tests/src/tests/client/resharding_v2.rs:925]: It's funny, those tests don't test resharding, they were just using the resharding testing framework.

**Review by wacban — APPROVED** [review]
LGTM, thanks!

File: chain/chunks/src/client.rs (hunk 1 above)
Comment (Trisfald) [reply chain/chunks/src/client.rs:123]: Yeah, thought so, maybe with some slightly changes

Let me know if I should keep this function alive

File: chain/client/src/sync/state.rs (hunk 3 above)
Comment (Trisfald) [reply chain/client/src/sync/state.rs:320]: Indeed, if I get to delete `ShardSyncStatus::ReshardingScheduling` this part will be gone as well

File: integration-tests/src/tests/client/resharding_v2.rs (hunk 4 above)
Comment (Trisfald) [reply integration-tests/src/tests/client/resharding_v2.rs:925]: This file was.. an interesting read 🙃

Comment (shreyan-gupta) [reply integration-tests/src/tests/client/resharding_v2.rs:925]: Oh, @Longarithm was using the test framework for some other type of tests

File: chain/chunks/src/client.rs (hunk 1 above)
Comment (wacban) [reply chain/chunks/src/client.rs:123]: Yeah, I think it's easier to keep it for now.

Comment (Trisfald) [reply chain/chunks/src/client.rs]: Done!

**Review by shreyan-gupta — APPROVED** [review]
Yay! Tiny bit scared that we might be deleting stuff we may need, but should be fine for now. We just need to keep the old implementation in mind.

**Pull request merged**